gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+      rnd.nextBytes(bytes);
+      writer.write(ByteBuffer.wrap(bytes));
+    }
+  }
+
+  @Test
+  public void testFileExists() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.exists(path)).isTrue();
+  }
+
+  @Test
+  public void testFileSize() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.size(path)).isEqualTo(SML_SIZE);
+  }
+
+  @Test
+  public void testReadByteChannel() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    long size = Files.size(path);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    ByteBuffer buf = ByteBuffer.allocate(1024 * 1024);
+    int timeout = 1024 * 1024;
+    int read = 0;
+    while (chan.isOpen()) {
+      if (timeout-- <= 0) {
+        break;
+      }
+      int rc = chan.read(buf);
+      if (rc < 0) {
+        // EOF
+        break;
+      }
+      buf.clear();
+      assertThat(rc).isGreaterThan(0);
+      read += rc;
+      assertThat(chan.position() == read);
+    }
+    assertThat(read).isEqualTo(size);
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(BIG_FILE);
+    long size = Files.size(path);
+    final ByteBuffer buf1a = ByteBuffer.allocate(100);
+    final ByteBuffer buf1b = ByteBuffer.allocate(100);
+    final ByteBuffer buf2a = ByteBuffer.allocate(100);
+    final ByteBuffer buf2b = ByteBuffer.allocate(100);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    reallyRead(chan, buf1a);
+    long dest = size / 2;
+    chan.position(dest);
+    reallyRead(chan, buf2a);
+    // now go back and read it again
+    // (we do 2 locations because 0 is sometimes a special case).
+    chan.position(0);
+    reallyRead(chan, buf1b);
+    chan.position(dest);
+    reallyRead(chan, buf2b);
+    // if the two spots in the file have the same contents, then this isn't a good file for this
+    // test.
+    assertThat(buf1a.array()).isNotEqualTo(buf2a.array());
+    assertThat(buf1a.array()).isEqualTo(buf1b.array());
+    assertThat(buf2a.array()).isEqualTo(buf2b.array());
+  }
+
+  @Test
+  public void testCreate() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(PREFIX + randomSuffix());
+    try {
+      // file shouldn't exist initially. If it does it's either because it's a leftover
+      // from a previous run (so we should delete the file)
+      // or because we're misconfigured and pointing to an actually important file
+      // (so we should absolutely not delete it).
```

Comment:
Oh oops, you're right of course I didn't mean for that to be inside the try! Good catch, thank you!